osu.Game.Rulesets.Taiko/UI/TaikoPlayfieldAdjustmentContainer.cs

```diff
@@ -51,10 +55,38 @@ protected override void Update()
             // Limit the maximum relative height of the playfield to one-third of available area to avoid it masking out on extreme resolutions.
             relativeHeight = Math.Min(relativeHeight, 1f / 3f);
 
-            Y = base_position;
-
             Scale = new Vector2(Math.Max((Parent!.ChildSize.Y / 768f) * (relativeHeight / base_relative_height), 1f));
             Width = 1 / Scale.X;
         }
+
+        public double ComputeTimeRange()
+        {
+            float currentAspect = Parent!.ChildSize.X / Parent!.ChildSize.Y;
+
+            if (LockPlayfieldAspectRange.Value)
+                currentAspect = Math.Clamp(currentAspect, MINIMUM_ASPECT, MAXIMUM_ASPECT);
+
+            // in a game resolution of 1024x768, stable's scrolling system consists of objects being placed 600px (widthScaled - 40) away from their hit location.
+            // however, the point at which the object renders at the end of the screen is exactly x=640, but stable makes the object start moving from beyond the screen instead of the boundary point.
+            // therefore, in lazer we have to adjust the "in length" so that it's in a 640px->160px fashion before passing it down as a "time range".
```

Comment:
what does this mean? what is "a 640px->160px fashion"?

Reply:
As in, the `inMsLength` property in stable currently gives the time spent by the object moving from 760px to 160px (760px is determined by `stable_hit_location + inLength`, when game resolution is 1024x768).

This is bad for lazer, because 640px in stable is considered the right edge of the playfield, i.e. stable begins scrolling the object from outside of the playfield, while lazer begins scrolling the object from the right edge of the playfield instead (technically scrolls from the right edge of the `HitObjectContainer`).

So to compromise this, the `inLength` is changed to be calculated by `widthScaled - stable_hit_location` (distance between right edge of playfield and hit location) instead of `widthScaled - 40` (stable's distance, 600px)